src/DispatchR/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -50,30 +50,24 @@ public static IServiceCollection AddDispatchR(this IServiceCollection services,
         var streamPipelineBehaviorType = typeof(IStreamPipelineBehavior<,>);
         var syncNotificationHandlerType = typeof(INotificationHandler<>);
 
+        var otherHandlerTypes = new HashSet<Type>()
```

Comment:
This change is not related to the fix, but I thought we can avoid constructing the handler types array multiple times.